src/checkbox/src/Checkbox.js

```diff
@@ -162,7 +73,7 @@ class Checkbox extends PureComponent {
           onChange={onChange}
           disabled={disabled}
           aria-invalid={isInvalid}
-          innerRef={this.handleInnerRef}
+          ref={ref}
```

Comment:
Now that we have `forwardRef` here, we drop support for the `handleInnerRef` -- is there a better way to go about this @mshwery

Reply:
Rule of thumb:

1. `Box` does not support `ref` yet. If adding a ref to a `Box` component you need to pass it to `innerRef={ref}`
2. If adding a ref to any internal Evergreen component we should use `ref` (they are all either `PureComponent` or `forwardRef` components).

Reply:
As for your question there are two ways to keep the behavior intact:

Keep the `handleInnerRef` shim:
```
  handleInnerRef = el => {	
    if (el) {	
      el.indeterminate = indeterminate	
    }

    // manually update the ref (depending on whether it's a legacy ref or `createRef`
    if (ref) { ... }
  }
```

Use `useEffect` to trigger an update to set the `inputInstance.indeterminate` value:
```js
useEffect((
  if (ref?.current && props.indeterminate) {
    ref.current.indeterminate = true
  }
), [ref, ref?.current, props.indeterminate])
```